python/lsst/analysis/tools/actions/plot/matrixPlot.py

```diff
@@ -0,0 +1,384 @@
+        "'x' and 'y'. This does not affect the visibility of minor tick labels. For example, setting this "
+        "field to ['x', 'y'] will hide both minor ticks.",
+        default=[],
+        maxLength=2,
+        itemCheck=lambda s: s in ["x", "y"],
+    )
+
+    dpi = Field[int](
+        doc="The resolution of the figure.",
+        default=300,
+    )
+
+    guideLines = ConfigDictField[str, guideLinesConfig](
+        doc="Dictionary of guide lines for the x and y axes. The keys are 'x' and 'y', and the values are "
+        "instances of `guideLinesConfig`.",
+        default={},
+        dictCheck=lambda d: all([k in ["x", "y"] for k in d]),
+    )
+
+    def getInputSchema(self) -> KeyedDataSchema:
+        base: list[tuple[str, type[Vector]]] = []
+        base.append((self.matrixKey, Vector))
+        return base
+
+    def __call__(self, data: KeyedData, **kwargs) -> Figure:
+        self._validateInput(data, **kwargs)
+        return self.makePlot(data, **kwargs)
+
+    def _validateInput(self, data: KeyedData, **kwargs: Any) -> None:
+        # Check that the input data contains all the required keys.
+        needed = set(k[0] for k in self.getInputSchema())
+        if not needed.issubset(data.keys()):
+            raise ValueError(f"Input data does not contain all required keys: {self.getInputSchema()}")
+        # Check the input data is a matrix, i.e. a 2d array.
+        if not isinstance(data[self.matrixKey], np.ndarray) and data[self.matrixKey].ndim != 2:
+            raise ValueError(f"Input data is not a 2d array: {data[self.matrixKey]}")
+        # Check that the keyword arguments are valid.
+        acceptableKwargs = {"plotInfo", "skymap", "band", "metric_tags", "fig"}
+        if not set(kwargs).issubset(acceptableKwargs):
+            raise ValueError(
+                f"Only the following keyword arguments are allowed: {acceptableKwargs}. Got: {kwargs}"
+            )
+
+    def makePlot(self, data: KeyedData, plotInfo: Mapping[str, str] | None = None, **kwargs: Any) -> Figure:
+        """
+        Plot a matrix of values.
+
+        Parameters
+        ----------
+        data : `~lsst.analysis.tools.interfaces.KeyedData`
+            The data to plot.
+        plotInfo : `dict`, optional
+            A dictionary of information about the data being plotted.
+        **kwargs
+            Additional keyword arguments to pass to the plot.
+
+        Returns
+        -------
+        fig : `~matplotlib.figure.Figure`
+            The resulting figure.
+        """
+        matrix = data[self.matrixKey]
+        # Calculate default limits only if needed.
+        if self.vmin is None or self.vmax is None:
+            default_limits = apViz.PercentileInterval(98.0).get_limits(np.abs(matrix.flatten()))
+        else:
+            default_limits = (None, None)
+
+        # Set the value range using overrides or defaults.
+        vrange = (self.vmin or default_limits[0], self.vmax or default_limits[1])
```

Comment:
I think you still need to put `default_limits[0] if self.vmin is None else self.vmin` in case `self.vmin == 0`, which evaluates as `False` (and same for vmax although I don't think vmax=0 would work very well if at all).

Reply:
Good point, done!